types/views/views.go

```diff
@@ -44,83 +44,77 @@ func (v *StringSlice) UnmarshalJSON(b []byte) error {
 }
 
 // IsNil reports whether the underlying slice is nil.
-func (v StringSlice) IsNil() bool { return v.ж == nil }
+func (v Slice[T]) IsNil() bool { return v.ж == nil }
 
 // Len returns the length of the slice.
-func (v StringSlice) Len() int { return len(v.ж) }
+func (v Slice[T]) Len() int { return len(v.ж) }
 
 // At returns the string at index `i` of the slice.
-func (v StringSlice) At(i int) string { return v.ж[i] }
+func (v Slice[T]) At(i int) T { return v.ж[i] }
 
 // AppendTo appends the underlying slice values to dst.
-func (v StringSlice) AppendTo(dst []string) []string {
+func (v Slice[T]) AppendTo(dst []T) []T {
 	return append(dst, v.ж...)
 }
 
 // AsSlice returns a copy of underlying slice.
-func (v StringSlice) AsSlice() []string {
+func (v Slice[T]) AsSlice() []T {
 	return v.AppendTo(v.ж[:0:0])
 }
 
 // IPPrefixSlice is a read-only accessor for a slice of netaddr.IPPrefix.
 type IPPrefixSlice struct {
-	// It is named distinctively to make you think of how dangerous it is to escape
-	// to callers. You must not let callers be able to mutate it.jd
-	ж []netaddr.IPPrefix
+	ж Slice[netaddr.IPPrefix]
 }
 
 // IPPrefixSliceOf returns a IPPrefixSlice for the provided slice.
-func IPPrefixSliceOf(x []netaddr.IPPrefix) IPPrefixSlice { return IPPrefixSlice{x} }
+func IPPrefixSliceOf(x []netaddr.IPPrefix) IPPrefixSlice { return IPPrefixSlice{SliceOf(x)} }
 
 // IsNil reports whether the underlying slice is nil.
-func (v IPPrefixSlice) IsNil() bool { return v.ж == nil }
+func (v IPPrefixSlice) IsNil() bool { return v.ж.IsNil() }
 
 // Len returns the length of the slice.
-func (v IPPrefixSlice) Len() int { return len(v.ж) }
+func (v IPPrefixSlice) Len() int { return v.ж.Len() }
 
-// At returns the IPPrefix at index `i` of the slice.
-func (v IPPrefixSlice) At(i int) netaddr.IPPrefix { return v.ж[i] }
+// At returns the string at index `i` of the slice.
```

Comment:
why'd this change to "string"?

Reply:
that's not intentional, I fixed it locally but then got distracted by the `staticcheck` and `go generate` failures